src/de/grobox/liberario/fragments/ProductDialogFragment.java

```diff
@@ -108,7 +110,28 @@ public void onClick(View v) {
 				// call listener if set
 				if(listener != null) {
 					EnumSet<Product> products = getProductsFromItems(adapter.getSelectedItems());
-					listener.onProductsChanged(products);
+                    // get the current network
+                    TransportNetwork current_network = Preferences.getTransportNetwork(getContext());
+					String network_id = current_network.getId().toString();
+                    SharedPreferences sp = getContext().getSharedPreferences(Preferences.NETWORK_SETTINGS, 0);
+                    SharedPreferences.Editor editor = sp.edit();
+                    String network_pref = network_id + "_selected_products";
+					if(products.size()==Product.values().length){
+                        // remove pref if all products are selected
+                        editor.remove(network_pref);
+                        editor.apply();
```

Comment:
I think we don't need to remove the preferences with the new approach. It is sufficient to only store them when the user changes them and otherwise leave them like they are.